tests/unit/lexicon/test_step.py

```
@@ -112,13 +117,72 @@ def test_rejects_a_shape_outside_the_union(cls):
+    def test_still_checks_the_components_of_an_entrypoint_that_only_routes():
+        @ritual("plain")
+        def _enter(components: Ping) -> Transition:
+            return done(Pong(n=components.n))
+
+        with pytest.raises(RitualBoundaryError):
+            asyncio.run(_enter.run(Elsewhere(n=1)))
+
+    # `def` is read off the value, not off the function, so a sync body that
+    # hands back a coroutine is awaited rather than mistaken for a transition.
+    @staticmethod
+    def test_awaits_a_coroutine_a_sync_body_hands_back():
```

Ruff (0.16.0)

[warning] Missing return type annotation for staticmethod `test_awaits_a_coroutine_a_sync_body_hands_back`

Add return type annotation: `None`

(ANN205)